go/types/set_iterator.go

```diff
@@ -0,0 +1,207 @@
+// Copyright 2016 Attic Labs, Inc. All rights reserved.
+// Licensed under the Apache License, version 2.0:
+// http://www.apache.org/licenses/LICENSE-2.0
+
+package types
+
+import (
+	"github.com/attic-labs/noms/go/d"
+)
+
+// SetIterator defines methods that can be used to efficiently iterate through a set in 'Noms-defined'
+// sorted order.
+type SetIterator interface {
+	// Next returns subsequent values from a set. It returns nil, when no objects remain
+	Next() Value
+
+	// NextFrom(v) advances the cursor to a new position based on the following rules and returns the value at that position.
+	// * if v <= curValue, advances to the next value in the iteration.
```

Comment:
I don't understand this case at all -- first of all 'curValue' is not defined in this interface anywhere, so I don't know what it refers to...